components/eventing-controller/reconciler/subscription/reconciler.go

```diff
@@ -0,0 +1,731 @@
+	}
+
+	// Sync the BEB Subscription with the Subscription CR
+	if statusChangedForBeb, err := r.syncBEBSubscription(subscription, &result, ctx, log, apiRule); err != nil {
+		log.Error(err, "error while syncing BEB subscription")
+		return ctrl.Result{}, err
+	} else {
+		statusChanged = statusChanged || statusChangedForBeb
+	}
+
+	if r.isInDeletion(subscription) {
+		// Remove finalizers
+		if err := r.removeFinalizer(subscription, ctx, log); err != nil {
+			return ctrl.Result{}, err
+		}
+		result.Requeue = false
+		return result, nil
+	}
+
+	// Save the subscription status if it was changed
+	if statusChanged {
+		if err := r.Status().Update(ctx, subscription); err != nil {
+			log.Error(err, "Update subscription status failed")
+			return ctrl.Result{}, err
+		}
+		result.Requeue = true
+	}
+
+	return result, nil
+}
+
+// syncFinalizer sets the finalizer in the Subscription
+func (r *Reconciler) syncFinalizer(subscription *eventingv1alpha1.Subscription, result *ctrl.Result, ctx context.Context, logger logr.Logger) error {
+	// Check if finalizer is already set
+	if r.isFinalizerSet(subscription) {
+		return nil
+	}
+	if err := r.addFinalizer(subscription, ctx, logger); err != nil {
+		return err
+	}
+	result.Requeue = true
+	return nil
+}
+
+// syncBEBSubscription delegates the subscription synchronization to the backend client. It returns true if the subscription status was changed.
+func (r *Reconciler) syncBEBSubscription(subscription *eventingv1alpha1.Subscription,
+
+	result *ctrl.Result, ctx context.Context, logger logr.Logger, apiRule *apigatewayv1alpha1.APIRule) (bool, error) {
+	logger.Info("Syncing subscription with BEB")
+
+	//No need to initialize in every sync
+	//r.bebClient.Initialize()
+
+	// if object is marked for deletion, we need to delete the BEB subscription
+	if r.isInDeletion(subscription) {
+		return false, r.deleteBEBSubscription(subscription, logger, ctx)
+	}
+
+	var statusChanged bool
+	var err error
+	if statusChanged, err = r.bebClient.SyncBebSubscription(subscription, apiRule); err != nil {
+		logger.Error(err, "Update BEB subscription failed")
+		condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscribed, eventingv1alpha1.ConditionReasonSubscriptionCreationFailed, corev1.ConditionFalse)
+		if err := r.updateCondition(subscription, condition, ctx); err != nil {
+			return statusChanged, err
+		}
+		return false, err
+	}
+
+	if !subscription.Status.IsConditionSubscribed() {
+		condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscribed, eventingv1alpha1.ConditionReasonSubscriptionCreated, corev1.ConditionTrue)
+		if err := r.updateCondition(subscription, condition, ctx); err != nil {
+			return statusChanged, err
+		}
+		statusChanged = true
+	}
+
+	statusChangedAtCheck, retry, errTimeout := r.checkStatusActive(subscription)
+	statusChanged = statusChanged || statusChangedAtCheck
+	if errTimeout != nil {
+		logger.Error(errTimeout, "Timeout at retry")
```

Comment:
An error message should start with a small letter

Reply:
Done